### PR TITLE
Add flac to base.yaml

### PR DESCRIPTION
**Pull request by 708yamaguchi**
I add `flac` package to base.yaml

Links:
arch: https://www.archlinux.org/packages/extra/x86_64/flac/
debian: https://packages.debian.org/buster/flac
fedra: https://apps.fedoraproject.org/packages/mingw-flac
gentoo: https://packages.gentoo.org/packages/media-libs/flac
ubuntu: https://packages.ubuntu.com/bionic/flac
@k-okada

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -800,6 +800,8 @@ ffmpeg2theora:
   fedora: [ffmpeg2theora]
   gentoo: [media-video/ffmpeg2theora]
   ubuntu: [ffmpeg2theora]
+flac:
```
Comment (k-okada) [inline rosdep/base.yaml:803]: please add fedra/gentoo ... and also you'd better to add more information to help maintainer's decision https://github.com/ros/rosdistro/pull/19652

Comment (708yamaguchi) [reply rosdep/base.yaml:803]: Thank you for your advice!
I updated the comment.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml [hunk 2]
```diff
@@ -800,6 +800,12 @@ ffmpeg2theora:
...
+  arch: [flac]
+  debian: [flac]
+  fedora: [mingw-flac]
```
[the remark this answers is not in the document]
Comment (708yamaguchi) [reply rosdep/base.yaml]: Thank you for your advice!
I did not known `mingw-flac` is Windows-compatible version.

However, your advised URL (https://apps.fedoraproject.org/packages/flac ) seems to be different from what I would like to install.
I think the upstream page should be https://xiph.org/flac/

Therefore, I do not add `flac` for Fedora.

[the remark this answers is not in the document]
Comment (clalancette) [reply rosdep/base.yaml]: > However, your advised URL (https://apps.fedoraproject.org/packages/flac ) seems to be different from what I would like to install.
> I think the upstream page should be https://xiph.org/flac/ (You can find this URL at the other distributions' package pages.)

I guess I don't understand what you are trying to achieve.  For all of the distributions, you are going to get the version of `flac` that is packaged in their archives (not the upstream one).  For this particular key, you are going to get the binaries `/usr/bin/flac` (for instance) on Debian and Ubuntu.  The `flac` package on fedora provides exactly the same thing.

[the remark this answers is not in the document]
Comment (708yamaguchi) [reply rosdep/base.yaml]: I am very sorry for your inconvenience.

> I guess I don't understand what you are trying to achieve.

I would like to install `flac` package, which describes "Free Lossless Audio Codec", by `rosdep install` command.

However in Fedora, flac package (https://apps.fedoraproject.org/packages/flac ) is not "Free Lossless Audio Codec" but "Flacon extracts individual tracks ..."
In other words, `flac` package in Fedora is different from `flac` package in the disctirubtions other than Fedora.

I think it is confusing to put these different packages together into one `flac` key.

[the remark this answers is not in the document]
Comment (708yamaguchi) [reply rosdep/base.yaml]: Sorry.

It seems that overview description in `flac` package in Fedora (https://apps.fedoraproject.org/packages/flac ) seems to be wrong.
The description is the same as the `falcon` package in Fedora (https://apps.fedoraproject.org/packages/flacon ).

The both packages say:
```
Flacon extracts individual tracks from one big audio file containing
the entire album of music and saves them as separate audio files.
To do this, it uses information from the appropriate CUE file.
Besides, Flacon makes it possible to conveniently revise or specify
tags both for all tracks at once or for each tag separately.
```

If so, I can understand adding `fedora: [flac]` line.

[the remark this answers is not in the document]
Comment (708yamaguchi) [reply rosdep/base.yaml]: Sorry again.

I found specification file for `flac` package in Fedora.
https://apps.fedoraproject.org/packages/flac/sources/spec/

It says:
```
%description
FLAC stands for Free Lossless Audio Codec. Grossly oversimplified, FLAC
is similar to Ogg Vorbis, but lossless. The FLAC project consists of
the stream format, reference encoders and decoders in library form,
flac, a command-line program to encode and decode FLAC files, metaflac,
a command-line metadata editor for FLAC files and input plugins for
various music players.
```

From this specification file, I believe the `flac` package in Fedora is what I would like to install.

I am sorry for troubling you.

[the remark this answers is not in the document]
Comment (clalancette) [reply rosdep/base.yaml]: Yeah, I think there is a bug in the Fedora package database; it often shows the wrong description for packages.

In any case, this looks good to me now.  Thank you for iterating.

**Review by clalancette — APPROVED** [review]

**Pull request merged**